### PR TITLE
Code cleanup, perf optimizations and added .gitignore

### PR DESCRIPTION
**Pull request by budgetdevv**
Lel

### CODE REVIEW

**Comment by DelightedCat** [comment]
Thank you for opening a pull request! I've reviewed your changes and I have some notes to make.

First of all, it seems that you've made some performance improvements, which are looking great. Especially for projects like a web server performance tweaks like this are always nice to have.

Secondly, it seems like the `.gitignore` is far more extended than necessary. There are a lot of entries that will end up unused, and I'm trying to keep this file as simple and short as possible, specifically for each project. Please limit the changes to the `.gitignore` as much as possible by placing a new commit where only necessary entries are in this file (e.g. add `.idea/` when you're using IntelliJ or `.vscode/` when you're using Visual Studio Code).

To add up on that, it seems like you've used `var` in multiple places instead of explicitly typing variables, although I'll be changing that later on when I'll get back to this project myself. This is my preferred way of declaring variables as I have the impression that it increases readability, although it's true that I didn't explicitly mention this.

To finish it all up, it appears to me that you've shortened some methods and replaced unused `public` accessors to `private`. Thanks for that!

In conclusion, please remove the excessive entries from the `.gitignore` and I'll merge this one in!

**Comment by budgetdevv** [comment]
I could re-fork the repo, and make the changes...which should undo all the modifications to "var" c;

**Comment by DelightedCat** [comment]
That'd be even better! Thanks in advance. Do you think you can add the changes to this pull request, or should I close this one and wait for a new one?

**Comment by budgetdevv** [comment]
I _should_ be able to force-commit to this one

**Comment by DelightedCat** [comment]
Looks like there are some merge conflicts caused by some changes I've made to the `.gitignore`, I'll resolve them once your PR is ready to be merged!

**Comment by DelightedCat** [comment]
I was able to merge your changes with the develop branch while fixing the `.gitignore` file. Now I'll go around fixing the `var` occurrences. Thank you for contributing!
